src/main/java/seedu/healthmate/ChatParser.java

```diff
@@ -0,0 +1,172 @@
+import seedu.healthmate.command.commands.SaveMealCommand;
+import seedu.healthmate.command.commands.ListCommandsCommand;
+import seedu.healthmate.command.commands.AddMealEntryCommand;
+import seedu.healthmate.command.commands.DeleteMealCommand;
+import seedu.healthmate.command.commands.DeleteMealEntryCommand;
+import seedu.healthmate.command.commands.MealMenuCommand;
+import seedu.healthmate.command.commands.UpdateUserDataCommand;
+import seedu.healthmate.command.commands.TodayCalorieProgressCommand;
+import seedu.healthmate.command.commands.HistoricCalorieProgressCommand;
+import seedu.healthmate.command.CommandMap;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Encapsulates the main logic of the application by parsing user input into objects
+ * and storing them respectively.
+ */
+public class ChatParser {
+
+    public static final String CALORIE_SIGNALLER = "/c";
+
+    private static Logger logger = Logger.getLogger(ChatParser.class.getName());
+    private MealEntriesList mealEntries;
+    private MealList mealOptions;
+    private final HistoryTracker historyTracker;
+
+    public ChatParser(){
+        Logging.setupLogger(logger, ChatParser.class.getName());
+        this.historyTracker = new HistoryTracker();
+        logger.log(Level.INFO, "Initializing HistoryTracker");
+        UI.printSeparator();
+        this.mealEntries = historyTracker.loadMealEntries();
+        logger.log(Level.INFO, "Loaded MealEntries");
+        this.mealOptions = historyTracker.loadMealOptions();
+        logger.log(Level.INFO, "Loaded MealOptions");
+        UI.printSeparator();
+    }
+
+    /**
+     * Reads in user input from the command line
+     * and initiates the parsing process steered by one-token and two-token-based user prompts.
+     */
+    public void run() {
+        // check for health goal file existence and create file if none exists
+        logger.log(Level.INFO, "Checking if user data exists");
+        User user = User.checkForUserData(this.historyTracker);
+
+        Scanner scanner = new Scanner(System.in);
+        String userInput = "";
+
+        while (!userInput.equals("bye")) {
+            logger.log(Level.INFO, "Getting next user input line");
+            userInput = scanner.nextLine().strip();
+            switch (userInput) {
+            case "bye":
+                logger.log(Level.INFO, "User closes application");
+                UI.printFarewell();
+                break;
+            default:
+                try {
+                    this.multiCommandParsing(userInput, user);
+                    logger.log(Level.INFO, "User input contains more than 1 token");
+                } catch (ArrayIndexOutOfBoundsException a) {
+                    logger.log(Level.WARNING, "Invalid command", a);
+                    UI.printReply("Invalid command", "Retry: ");
+                }
```

Comment:
If there are no more potential cases to be added, you could consider just use a simple ```if...else...```. ```switch``` will be more suitable if there are more cases.